plugins/advisors/vulnerable-code/src/main/kotlin/VulnerableCode.kt

```diff
@@ -93,23 +93,35 @@ class VulnerableCode(override val descriptor: PluginDescriptor, config: Vulnerab
     override suspend fun retrievePackageFindings(packages: Set<Package>): Map<Package, AdvisorResult> {
         val startTime = Instant.now()
 
-        val purls = packages.mapNotNull { pkg -> pkg.purl.ifEmpty { null } }
-        val chunks = purls.chunked(BULK_REQUEST_SIZE)
+        // Workaround for #9298
+        // Create a list of pairs, where the first element is the purl without percent encoding
```

Comment:
No, that unfortunately won't work, because the ORT purls contain a mixture of "/" and "%2F" like in pkg:golang/github.com%2Fquic-go%2Fquic-go@0.40.0, and once you have replaced all "%2F" with "/" there is no more way back to the "mixed" ORT purl. This was the reason for having this implementation as pairs, where the first element is the modified purl, and the second element the original ORT purl.

Reply:
> there is no more way back to the "mixed" ORT purl.

Correct, but my point is that my implementation proposal should not require you to go back, as the VC-style-PURL is only used for the lookup in the VC results.

Anyway, we're now having a larger discussion about PURL encoding over [here](https://github.com/package-url/purl-spec/pull/176), and I'm tempted to say we should solve this completely differently: By not encoding `/` that occurs as part of the namespace at all, but only in the name.